File: Manuel/2025-08-26/오르막길/2846 - 2.cpp
#include <iostream>
#include <vector>
#define endl "\n"

using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> heights(n);

    for(int i = 0; i < n; i++)
    {
        cin >> heights[i];
    }

    int max_uphill = 0;
    int current_uphill = 0;

    for(int i = 0; i < n - 1; i++)
    {
        if(heights[i] < heights[i + 1])
        {
            current_uphill += heights[i + 1] - heights[i];
            max_uphill = max(max_uphill, current_uphill);
        }
        else
        {
            current_uphill = 0;
        }
    }
    cout << max_uphill << endl;
    return 0;
}

File: Manuel/2025-08-26/오르막길/2846 - 2.py
n = int(input())
heights = list(map(int, input().split()))

uphills = []
current_uphill = 0

for i in range(n - 1):
    if heights[i] < heights[i + 1]:
        current_uphill += heights[i + 1] - heights[i]
    else:
        uphills.append(current_uphill)
        current_uphill = 0
    
uphills.append(current_uphill)

print(max(uphills) if uphills else 0)